rust/lance-index/src/vector/ivf/builder.rs

```diff
@@ -40,8 +41,15 @@ pub struct IvfBuildParams {
 
     pub sample_rate: usize,
 
+    /// Precomputed partitions file (row_id -> partition_id)
+    /// mutually exclusive with `precomputed_shuffle_buffers`
     pub precomputed_partitons_file: Option<String>,
 
+    /// Precomputed shuffle buffers (row_id -> partition_id, pq_code)
+    /// mutually exclusive with `precomputed_partitons_file`
+    /// requires `centroids` to be set
+    pub precomputed_shuffle_buffers: Option<(Path, Vec<String>)>,
```

Comment:
So each buffer is 1 local file, IIUC?

could you specify what are `Path` and `Vec<String>` more explicitly. Not clear from the comments.

Reply:
it's multiple. because we might need multiple processes creating these buffers